jdaviz/core/astrowidgets_api.py

```diff
@@ -0,0 +1,523 @@
+import sys
+
+import numpy as np
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+from astropy.utils.introspection import minversion
+from astropy.wcs import NoConvergence
+from echo import delay_callback
+from glue.config import colormaps
+from glue.core import Data
+
+from jdaviz.configs.imviz.helper import data_has_valid_wcs, get_top_layer_index
+from jdaviz.core.events import SnackbarMessage
+
+__all__ = ['AstrowidgetsImageViewerMixin']
+
+ASTROPY_LT_4_3 = not minversion('astropy', '4.3')
+
+
+class AstrowidgetsImageViewerMixin:
+    """This class implements ``astrowidgets`` API for Jdaviz *image* viewer.
+    This does not provide a fully functional viewer, but rather
+    should be used as mixin into an existing viewer subclass.
+
+    A child class that uses this must run :meth:`init_astrowidgets_api`
+    within its ``__init__``.
+
+    .. note:: Do not use this for spectral or cube viewers.
+
+    """
+    RESERVED_MARKER_SET_NAMES = ['all']
+
+    # __init__ not called, so use this to setup.
+    def init_astrowidgets_api(self):

```

Comment:
One could argue I should try `__new__` but MRO gives me headache, so maybe we can do that after the release, if people have strong feeling about this. Such a change will not be user-facing anyway.

Reply:
I have a moderately strong feeling that it would indeed be better to use `__new__` or perhaps one of those fancy things in recent pythons that are sort of like "simpler metaclasses". But I agree that's not necessary given this is working right now!

Reply:
OK, let's leave it alone for now. I opened #871 so we don't forget about it.